src/metabase/api/dataset.clj

```diff
@@ -83,25 +84,37 @@
      (api/defendpoint POST [\"/:export-format\", :export-format export-format-regex]"
   (re-pattern (str "(" (str/join "|" (map u/qualified-name (qp.streaming/export-formats))) ")")))
 
+(def ^:private column-ref-regex #"\[\"ref\",\[(?:\"field\",\d+,null|\"expression\",\".+\")\]\]")
```

Comment:
This seems like it could be a little too strict... the column ref could be something like 

```clj
[:ref [:field 1 {:temporal-unit :month}]]
```

(serialized as JSON) which wouldn't match this regex. Maybe just check if starts and ends with square brackets? Pretty sure that will never be a raw column name.


wouldn't it be easier to parse the JSON column key than to try